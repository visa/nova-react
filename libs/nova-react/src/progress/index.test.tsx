/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import Progress from '.';

describe('Progress', () => {
  it('should render defaults correctly (linear)', async () => {
    const { container } = render(<Progress />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
    expect(container.firstElementChild?.className).toBe('v-progress v-progress-bar');
    expect(container.firstElementChild?.tagName).toBe('PROGRESS');
  });

  it('should render circular defaults correctly', async () => {
    const { container } = render(<Progress aria-label="default circular progress" progressType="circular" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
    expect(container.firstElementChild?.className).toBe('v-progress v-progress-circular');

    expect(container.firstElementChild?.tagName).toBe('DIV');
  });

  it('should pass props to progress linear correctly', () => {
    const { container } = render(
      <Progress
        className="test-class"
        completed
        invalid
        paused
        progressType="linear"
        style={{ width: 20 }}
        value={50}
      />
    );
    expect(container.firstElementChild?.className).toBe(
      'v-progress v-progress-bar v-progress-error v-progress-complete test-class'
    );
    expect(container.firstElementChild?.getAttribute('style')).toBe('animation-play-state: paused; width: 20px;');
  });

  it('should pass props to progress circular correctly', () => {
    const { container } = render(
      <Progress
        className="test-class"
        indeterminate
        paused
        progressSize={2}
        progressType="circular"
        style={{ width: 20 }}
        value={50}
      >
        Test
      </Progress>
    );
    expect(container.firstElementChild?.className).toBe(
      'v-progress v-progress-circular v-progress-indeterminate test-class'
    );
    expect(container.firstElementChild?.getAttribute('style')).toBe(
      'animation-play-state: paused; --v-progress-circular-size: 2px; width: 20px;'
    );
  });
});
