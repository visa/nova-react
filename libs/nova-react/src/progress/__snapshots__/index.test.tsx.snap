// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render circular defaults correctly 1`] = `
<div>
  <div
    aria-label="default circular progress"
    class="v-progress v-progress-circular"
    role="progressbar"
    style="animation-play-state: running; --v-progress-circular-size: 72px;"
  >
    <svg
      class="v-progress-circular-track"
    >
      <circle
        class="v-progress-circular-background"
      />
      <circle
        class="v-progress-circular-bar"
        style="stroke-dasharray: 0, 213.62830044410595; stroke-dashoffset: 0;"
      />
    </svg>
  </div>
</div>
`;

exports[`Progress should render defaults correctly (linear) 1`] = `
<div>
  <progress
    class="v-progress v-progress-bar"
    style="animation-play-state: running;"
  />
</div>
`;
